detectors/aws/ecs/test/ecs_test.go

```diff
@@ -91,6 +91,120 @@ func TestDetectV4LaunchTypeEc2(t *testing.T) {
 	assert.Equal(t, expectedResource, res, "Resource returned is incorrect")
 }
 
+// successfully returns resource when process is running on Amazon ECS environment
+// with Metadata v4 with the EC2 Launch type and bad ContainerARN.
+func TestDetectV4LaunchTypeEc2BadContainerArn(t *testing.T) {
+	testServer := httptest.NewServer(http.HandlerFunc(func(res http.ResponseWriter, req *http.Request) {
+		if strings.HasSuffix(req.URL.String(), "/task") {
+			content, err := os.ReadFile("metadatav4-response-task-ec2.json")
+			if err == nil {
+				_, err = res.Write(content)
+				if err != nil {
+					t.Fatal(err)
+				}
+			}
+		} else {
+			content, err := os.ReadFile("metadatav4-response-container-ec2-bad-container-arn.json")
+			if err == nil {
+				_, err = res.Write(content)
+				if err != nil {
+					t.Fatal(err)
+				}
+			}
+		}
+	}))
+	defer testServer.Close()
+
+	os.Clearenv()
+	_ = os.Setenv(metadataV4EnvVar, testServer.URL)
```

Comment:
Don't we have test helpers for setting environment variables without stepping on other tests?

Looks like [these](https://github.com/open-telemetry/opentelemetry-go/blob/main/internal/internaltest/env.go) may only be available in the core repo.  It would be good to be able to use them in this repo as well.  @pellared can you create an issue to track this if you agree?

Reply:
The tests could simply use https://pkg.go.dev/testing#T.Setenv